test/remote-connector.test.js

```diff
@@ -87,3 +87,81 @@ describe('RemoteConnector', function() {
     });
   });
 });
+
+describe('Custom Path', function() {
+
+  var test = this;
+
+  before(function(done) {
+
+    test.server = server = loopback();
+    server.set('legacyExplorer', false);
+    server.use('/api', loopback.rest());
+    server.set('host', '127.0.0.1');
+    server.set('port', 3001);
+    server.set('restApiRoot', '/api');
+    
+    var db_config = {
+      connector: loopback.Memory,
+      name: 'db'
+    };
+    var db = loopback.createDataSource('db', db_config);
+
+    var ServerModel_config = {
+      dataSource: 'db',
+      http: {
+        path: '/custom'
+      }
+    };
+
+    var ServerModel = loopback.PersistedModel.extend('TestModel', {foo: 'string'}, ServerModel_config);
+    ServerModel.attachTo(db);
+
+    server.dataSource('db', db_config);
+
+    server.model(ServerModel);
+
+    test.client = client = loopback();
+    client.set('legacyExplorer', false);
+    client.use('/api', loopback.rest());
+    client.set('host', '127.0.0.1');
+    client.set('port', 3000);
+    client.set('restApiRoot', '/api');
```

Comment:
> is there any disadvantage to doing this?

It makes it more difficult to see what part of test setup is important/relevant to the test and what is just extra random additions.

In general, tests should use the most minimal setup that reproduces the problem.

> If I am not mistaken, the ServerModel and the RemoteModel should share the same class name, correct?

Yes, I think so.

> If they share a class name but have different DataSources, I should create two separate app instances. I am not entirely sure.

As I said before, you don't need an app - you can create client model & datasource without a client app, just keep them "floating around".

However, if it makes it easier to understand what's going on, then I can live with a client app. Just don't configure the REST transport for it, because you are not using it in your test.

```js
test.client = client = loopback();
// that's it, nothing else is needed
```